packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -0,0 +1,200 @@
+      const key: string = individualItemLayout.w.toString() + individualItemLayout.h.toString();
+      let cardSize = CardSize.small;
+      // recreating layout based off width and height of card. The width and height values are returned by RGl
+      // medium wide is 2 scale wide and 1 scale in height
+      // large is 2 scale high and wide
+      // mediumTall is 1 scale wide and 2 scales high
+      if (individualItemLayout.h === 1) {
+        cardSize = CardSize.section;
+      } else if (key === '28') {
+        cardSize = CardSize.large;
+      } else if (key === '18') {
+        cardSize = CardSize.mediumTall;
+      } else if (key === '24') {
+        cardSize = CardSize.mediumWide;
+      }
+      const itemLayout: IDashboardCardLayout = {
+        i: individualItemLayout.i!,
+        x: individualItemLayout.x,
+        y: individualItemLayout.y,
+        size: cardSize
+      };
+      newLayout.lg!.push(itemLayout);
+    });
+    if (newLayout !== this.state.layout) {
+      if (this.props.onLayoutChange) {
+        this.props.onLayoutChange(newLayout);
+      }
+      this.setState({
+        layout: newLayout
+      });
+    }
+  };
+
+  private _addCard = (cardId: string): void => {
+    const addCardPanelCards = this.state.cardsForAddCardPanel;
+    let cardIndex: number = -1;
+    const layout: DashboardGridBreakpointLayouts = this.state.layout;
+    // find the card selected in the list of cards in add card panel
+    addCardPanelCards.map((card: IDGLCard, index: number) => {
+      if (card.id === cardId) {
+        cardIndex = index;
+        // calculate new layout for the selected card from add card panel
+        const calculatedLayout: { x: number; y: number } = this._calculateNextCardPostion(card.cardSize);
+        const cardLayout: IDashboardCardLayout = { i: card.id, x: calculatedLayout.x, y: calculatedLayout.y, size: card.cardSize };
+        layout.lg!.push(cardLayout);
+      }
+    });
+    if (cardIndex !== -1) {
+      // remove the selected card from the add card panel and add it to the list of cards that are to be show in layout
+      const cardSelected = addCardPanelCards.splice(cardIndex, 1);
+      const layoutCards: IDGLCard[] = this.state.cardsForLayout;
+      layoutCards.push(cardSelected[0]);
+      this.setState({
+        cardsForAddCardPanel: addCardPanelCards,
+        cardsForLayout: layoutCards,
+        layout: layout
+      });
+      // scroll to the card that was added to the layout
+      setTimeout(() => {
+        if (document.getElementById(cardId + 'dglCard')) {
+          document.getElementById(cardId + 'dglCard')!.scrollIntoView({ behavior: 'smooth' });
+          const css = getClassNames(getCardStyles!);
+          document.getElementById(cardId + 'dglCard')!.classList.add(css.fadeIn);
+        }
+      }, 100);
+    }
+  };
+
+  // calculate the position for the selected card from add card panel
+  private _calculateNextCardPostion = (cardSize: CardSize) => {
```

Comment:
>_calculateNextCardPostion [](start = 10, length = 25)

please write a better comment on what logic we are following

Reply:
added comments